Small intro into Indexes (#18439)
yugabyte/yugabyte-db: docs/content/preview/explore/indexes-constraints/_index.md

type: indexpage
---

> Indexes in YugabyteDB are stored in a distributed manner just like tables. They are split into tablets and replicated. Updates to indexes are transactional which means that row updates and the corresponding index updates occur as a single transaction. Similar to tables, they are stored in [LSM](https://en.wikipedia.org/wiki/Log-structured_merge-tree) format as compared to the [B-tree](https://www.postgresql.org/docs/current/btree-implementation.html#BTREE-STRUCTURE) structure in PostgreSQL.

Reviewer: ```suggestion
As with tables, indexes in YugabyteDB are stored in a distributed manner - that is, they are split into tablets and replicated. Updates to indexes are transactional, which means that row updates and the corresponding index updates occur as a single transaction. Similar to tables, they are stored in [LSM](https://en.wikipedia.org/wiki/Log-structured_merge-tree) format, as opposed to the [B-tree](https://www.postgresql.org/docs/current/btree-implementation.html#BTREE-STRUCTURE) structure used by indexes in PostgreSQL.
```

> YugabyteDB supports most of the PostgreSQL semantics in [YSQL API](../../api/ysql/) and in [YCQL API](../../api/ycql/), most of Cassandra index semantics are supported including other improvements.

Reviewer: ```suggestion
YugabyteDB supports most of the PostgreSQL index semantics in the [YSQL API](../../api/ysql/), and the [YCQL API](../../api/ycql/) supports most of the Cassandra index semantics while incorporating other improvements.
```
